icu4c/source/test/intltest/msfmrgts.cpp

```diff
@@ -987,11 +988,29 @@ void MessageFormatRegressionTest::Test4112104()
     delete format;
 }
 
+void MessageFormatRegressionTest::TestICU12584() {
+    // 42 placeholders.
+    // We were able to handle only 10 due to the bug ICU-12584.
+     UnicodeString pattern("{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}" \
+        "{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}{3}{1}{2}" \
+        "{3}{1}{2}{3}");
+
+    UErrorCode status = U_ZERO_ERROR;
+    int count;
```

Comment:
We use specific types in our API, and so should use them in test code when we call such API.

int --> int32_t

Unfortunately, you should then use a cast `(int)count` in the logln() because the `%d` specifically wants an int.

Reply:
done